contracts/integrations/test/simulation/stake_management_test.ts

```diff
@@ -0,0 +1,53 @@
+import { BlockchainBalanceStore } from '@0x/contracts-exchange';
+import { blockchainTests } from '@0x/contracts-test-utils';
+import * as _ from 'lodash';
+
+import { Staker } from '../actors';
+import { DeploymentManager } from '../utils/deployment_manager';
+import { AssertionResult } from '../utils/function_assertions';
+
+import { PoolManagementSimulation } from './pool_management_test';
+import { Simulation, SimulationEnvironment } from './simulation';
+
+export class StakeManagementSimulation extends Simulation {
+    protected async *_assertionGenerator(): AsyncIterableIterator<AssertionResult | void> {
+        const { deployment, balanceStore } = this.environment;
+        const poolManagement = new PoolManagementSimulation(this.environment);
+
+        const staker = new Staker({ name: 'Staker', deployment, simulationEnvironment: this.environment });
+        await staker.configureERC20TokenAsync(deployment.tokens.zrx);
+        balanceStore.registerTokenOwner(staker.address, staker.name);
+
+        const actions = [
+            staker.simulationActions.validStake,
+            staker.simulationActions.validUnstake,
+            staker.simulationActions.validMoveStake,
+            poolManagement.generator,
+        ];
+        while (true) {
+            const action = _.sample(actions);
+            yield (await action!.next()).value; // tslint:disable-line:no-non-null-assertion
+        }
+    }
+}
+
+blockchainTests.skip('Stake management fuzz test', env => {
```

Comment:
Could be nice to have a way to run this w/o recompiling. Like, `yarn fuzz`

Reply:
I'm encountering some weird shit with mocha when trying to do this 😕 
but pls do add to your PR if you can figure it out